schemas/EGA.ArrayAssay.json

```diff
@@ -2,8 +2,8 @@
     "$schema": "https://json-schema.org/draft/2019-09/schema",
     "$id": "https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas/EGA.ArrayAssay.json",
     "type": "object",
-    "title": "EGA ArrayAssay metadata schema",
-    "description": "Metadata schema used by the European Genome-phenome Archive (EGA) to validate its ArrayAssay metadata object. This object is intended to contain metadata about the physical array to which the assayed molecule is hybridized. Used exclusively an Array Format (AF) submission. Further details can be found in the EGA-metadata-schema GitHub repository (https://github.com/EbiEga/ega-metadata-schema/tree/main/schemas) and EGA-archive website (https://ega-archive.org/)",
+    "title": "EGA ArrayAssay metadata schema v0.0.1",
```

Comment:
Consider a separate attribute, e.g. `version`, that can track the version instead of appending this to the title. This way, the title can be re-used in other place w/o having the version number exposed. Most users won't know/care about what version they are using, only that they want to use whatever is the "latest" or "current" version.

Reply:
I agree. I'll take a look at ``jsonschema2md`` to see if they use some keyword to store versions, but if not I'll simply use ``version``.

Reply:
@malloryfreeberg I'm wondering what the best choice is:

1. Using semantic versioning **within the ``$id``** of each schema. 
2. Using semantic versioning **outside of the ``$id``**. 

Each has its own advantages and disadvantages. (1) Would mean that we would need to change all references within each schema to other schemas (e.g. ``EGA.sample.json`` referencing ``EGA.common_definitions.json``) every time their versions would change: pointer to ``v0.0.1`` of ``EGA.common_definitions.json`` would be different from point of ``v0.0.2``.  

Nevertheless, with this approach (the former) we would **always** keep pointers consistently between versions, so ``v0.0.1`` of a schema could point to ``v0.0.3`` of another without having to be the last version residing within our repo. In other words, we would have to release the same "compatible" versions together and for them to be used independently one would have to download the release and isolate it in a directory. 

On top of it, the ``$id``, even if not required to be URL-resolvable, should point to our resources within GH. If we use the semantic versioning within this field we would need to create folders for each release if we want to keep such pointers. 

As references:

- **ENA** seems to have one functioning version of [each schema within GH](https://github.com/enasequence/schema/tree/master/src/main/resources/uk/ac/ebi/ena/sra/schema) and that's it. Their **references are static**, independent of versions. For example:
````
<xs:schema xmlns:com="SRA.common" xmlns:xs="http://www.w3.org/2001/XMLSchema">
    <xs:import namespace="SRA.common" schemaLocation="SRA.common.xsd"/>
...
     <xs:extension base="com:ObjectType">
````
- **HCA** does the opposite, having [schemas in bundles](https://schema.humancellatlas.org/a), and those **versions within each schema's ``$id``**. For example: ````https://schema.humancellatlas.org/bundle/5.2.1/submission#/definitions/submitted_files````

For now I'll keep the references static, less chaos and easier to adapt to GH URLs with our ``$id``s, but we may want to switch at some point.

Reply:
I agree with your assessment and recommendation. Indeed, it was quite confusing to set up in HCA all the schemas with versions in their URLs (hence why we had a separate step to publish them), but I think this is overkill for us at the moment.